text/3263-pre-release-sticky.md

```diff
@@ -0,0 +1,117 @@
+- Feature Name: `pre-release-sticky`
+- Start Date: 2022-05-10
+- RFC PR: [rust-lang/rfcs#3263](https://github.com/rust-lang/rfcs/pull/3263)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+The following assume:
+
+ - A release is a version without a "pre-release" tag (like `1.0.0`)
+ - A pre-release is a version with a "pre-release" tag (like `1.0.0-alpha`)
+
+Cargo will now apply no precedence rule on pre-release by default, this mean that `1.0.0-alpha.0` will be interpreted as `=1.0.0-alpha.0` by cargo instead of `^1.0.0-alpha.0`.
```

Comment:
Well, since I wanted this to be transparent I didn't include the opt-in alternative. The problem of opt-in is that this will help no one. User can already "fix" this problem by using `=` but they don't cause no one expect this problem. The real fix is to change the default behaviour. Also the opt-in solution could introduce a problem is this recursive other all dependencies ? Cause if this is only for the current crate it's almost useless.

Reply:
I don't think that will work, see https://github.com/rust-lang/rfcs/pull/3263#discussion_r868928119.